jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -184,24 +185,23 @@ public boolean contains(Point2D.Double p, double scaleDenominator) {
               AttributeKeys.getStrokeTotalWidth(this, scaleDenominator)
                   * attr().get(STROKE_MITER_LIMIT));
       if (gs.createStrokedShape(path).contains(p)) {
-        return true;
+        result = true;
       } else {
         if (isClosed()) {
           return false;
         }
       }
-    }
-    if (!isClosed()) {
+    } else if (!isClosed()) {
```

Comment:
Great work on this update! I noticed in the committed changes that you've implemented some key improvements. One thing that stood out to me was the optimization in the else method used. It's clear that these changes could significantly enhance the performance and readability of the code. However, I'd recommend adding some additional comments or documentation around the more complex sections for clarity. This would make it easier for future contributors to understand the logic behind these changes. Keep up the good work!